crates/autopilot/src/solvable_orders.rs

```diff
@@ -367,12 +392,9 @@ impl SolvableOrdersCache {
     }
 }
 
-/// Filters all orders whose owners or receivers are in the set of "banned"
+/// Finds all orders whose owners or receivers are in the set of "banned"
 /// users.
-async fn filter_banned_user_orders(
-    mut orders: Vec<Order>,
-    banned_users: &banned::Users,
-) -> Vec<Order> {
+async fn find_banned_user_orders(orders: &[Order], banned_users: &banned::Users) -> Vec<OrderUid> {
     let banned = banned_users
         .banned(orders.iter().flat_map(|order| {
             [
```

Comment:
Could you adjust the `banned users` filtering slightly as well in this PR.
Just noticed that we are currently running a lot of duplicate requests for checking the address `0x00000` due to the `receiver.unwrap_of_default()`.

I would suggest something like this to avoid these duplicate requests:
```rust
            orders
                .iter()
                .flat_map(|order| std::iter::once(order.metadata.owner).chain(order.data.receiver)),
```

Reply:
> a lot of duplicate requests

I think we don't make a lot of the requests since HasSet is used later. But removing 0x address makes sense.
https://github.com/cowprotocol/services/blob/488fb8b8db2641760d30c8aab6b1083956e81ac7/crates/order-validation/src/banned.rs#L56-L67